### PR TITLE
feat(objectives): harden weight validation — nan/inf, dominance guard

### PR DESCRIPTION
**Pull request by IsraelTraigent**
## Summary
- Fix bug where `float('nan')` slipped past `weight <= 0` check (nan comparisons return False)
- Add `math.isfinite` guard to reject nan, inf, -inf weights in `ObjectiveDefinition`
- Add multi-objective dominance guard: if >1 objective exists, no single one can hold 100% of normalized weight (prevents degenerate configs like `[1000000, 0.0001]`)
- Handle `None` weight in `from_dict` (was causing TypeError instead of defaulting to 1.0)
- Add info logging when weights are auto-normalized to sum to 1.0

## Test plan
- [x] 10 new tests in `TestWeightValidationHardening` class covering nan/inf, dominance, null weight, backward compat
- [x] All 36 existing objective tests pass (no regressions)
- [x] All 21 edge case tests pass
- [x] Backward compatible: arbitrary positive weights (70.0, 30.0, 2.0) still accepted and normalized
- [x] Single-objective with weight=1.0 still valid

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: traigent/core/objectives.py [hunk 1]
```diff
@@ -255,6 +259,32 @@ def from_objectives(
...
+            for obj in objectives:
+                nw = weights_normalized[obj.name]
+                if nw > 1.0 - 1e-9:
+                    raise ValueError(
+                        f"In a multi-objective schema, no single "
+                        f"objective can have 100% of the weight. "
+                        f"Objective '{obj.name}' has normalized "
+                        f"weight {nw:.6f}. Adjust weights so all "
+                        f"objectives contribute."
+                    )
```
[the remark this answers is not in the document]
Comment (IsraelTraigent) [reply traigent/core/objectives.py]: ixed in d7efe0f. Moved the dominance guard from from_objectives into ObjectiveSchema.__post_init__ so it fires on all construction paths, including from_dict deserialization. Added test_dominance_guard_on_deserialization to cover this.
